### ISSUE TITLE
 sentieon-nextflow job error in AWS batch

### ISSUE THREAD

**Issue by jo077** [goldfinchbio/sentieon-nextflow#2]
Hi All,

I am trying to use sentieon-nextflow in AWS batch. I have followed the step as mentioned in README.md. But I am getting following error.
Caused by:
  Task failed to start - CannotCreateContainerError: Error response from daemon: invalid volume specification: 'null:null:ro': invalid mount config for type "volume": invalid mount path: 'null' mount path must be absolute

Could you please let me know how to resolve this error?

Any help would be highly appreciated.

Thank you!

**Comment by sarthak1287** [comment]
This seems like an issue with AWS Batch Job definition you are creating. Please look at "ContainerProperties" -> MountPoints and Volumes configuration.

It should be something like below (Cloudformation) template:

      ContainerProperties:
        MountPoints:
          - ReadOnly: false
            SourceVolume: docker_scratch
            ContainerPath: /scratch       # Path you want to refer inside the container.
        Volumes:
          - Host:
              SourcePath: /<EC2_path_which_you_want_to_attach_as_volume_to_docker_container>
              Name: docker_scratch

If this seems correct , check the docker container you are using and confirm if those volumes/paths you want to attach are the same ones inside docker.

Regards,
Sarthak Patel

**Comment by jo077** [comment]
Hi Sarthak,

I followed as you have suggested. 

ContainerProperties:
    MountPoints:
       - ReadOnly: false
        SourceVolume: docker_scratch
         ContainerPath: /home      # Path you want to refer inside the container.
     Volumes:
       - Host:
           SourcePath: /home/ubuntu
           Name: docker_scratch

but every time I run the path is changing automatically to

ContainerProperties:
     MountPoints:
       - ReadOnly: false
         SourceVolume: aws-cli
         ContainerPath: /       # Path you want to refer inside the container.
     Volumes:
       - Host:
           SourcePath: /
           Name: aws-cli 
Could you please suggest me where why it is changing like this?

**Comment by sarthak1287** [comment]
Hello @jo077 ,

This is strange. Was your batch job definition created by NF or you created it manually using cloudformation/cdk ?

Kindly confirm if the AMI the batch is using is an ubuntu AMI otherwise "ubuntu" folder wont be available. 

Also, please can you set the min & desired vcpus on Batch Compute to "2" and login to corresponding EC2 (which gets spun up) and try to run the docker container you  want manually. 

In order to login, you will either need to specify the key in launch template/batch compute conf or add policy "AmazonSSMManagedInstanceCore" to IAM role so you can login through EC2 UI itself.

Regards,
Sarthak

**Comment by jo077** [comment]
Actually I have created job definition manually. 
Yes, AMI is a n ubuntu OS AMI.

vcpu is set to 2 and memory is 2048 mb.

I am using docker as mentioned here https://github.com/Sentieon/sentieon-docker.

Thank you!

**Comment by sarthak1287** [comment]
Are you able to run the docker image manually using "docker run -v ... -it ..." command and execute the code you want to run?

Regards,
Sarthak

**Comment by jo077** [comment]
Hi, when I am running docker manually it is running with command.

**Comment by sarthak1287** [comment]
Please can you upload the code and steps you are running in your personal repo and provide me access to the repo on github itself. That will be easiest to troubleshoot.

**Comment by jo077** [comment]
Okay, sure. I will do that and let you know.

**Comment by jo077** [comment]
Hi Sarthak, I have created the repository and sent you the invitation.

Thank you!

**Comment by jo077** [comment]
Hi All,

Finally able to resolve the issue. The steps I followed are given below.

- Install awscli out side /usr/local/bin for your docker image (I have used conda)
-  Mention the path for awscli in "aws.batch.cliPath =" in nextflow.config file
- As per your requirements set memory and vCPU in the job definition.
- In the sentieon_dnaseq.nf file give desired cpu and memory (e.g. In my case my input files were around ~17gb, so for mapping I gave 30 vCPU and 50gb in mapping process) .

Thank you!
